client/client_test.go

```diff
@@ -196,16 +196,12 @@ func TestParseHostURL(t *testing.T) {
 }
 
 func TestNewEnvClientSetsDefaultVersion(t *testing.T) {
-	env := envToMap()
-	defer mapToEnv(env)
-
-	envMap := map[string]string{
+	defer env.PatchAll(t, map[string]string{
```

Comment:
I noticed `env.Patch()` and `env.PatchAll()` have a difference in behaviour; where `env.Patch()` actually _patches_ the current environment variables, `env.PatchAll()` is _overriding_ the current environment (i.e., it removes all existing environment variables); https://github.com/gotestyourself/gotestyourself/blob/511344eed30e4384f010579a593dfb442033a692/env/env.go#L40-L45

I wonder if that function needs to have a different name, or if the `os.ClearEnv()` should be removed

Reply:
Yes, that is exactly how they are documented: https://godoc.org/github.com/gotestyourself/gotestyourself/env

> Patch changes the value of an environment variable

> PatchAll sets the environment to env

-----

> I wonder if that function needs to have a different name,

[All](http://www.thesaurus.com/browse/all?s=t) -> "the whole of", complete, entire

I think the name should imply that it replaces everything. What did you expect `PatchAll` to do?

Reply:
The “Patch” part lead me to believe I’d be modifying just what I sent, but keep other values as-is; even more because the “Patch” does so for a single var.

idk, “SetEnv()” / “Override()” would perhaps  perhaps be closed to that it does

Reply:
I think `SetEnv` would be too close to https://golang.org/pkg/os/#Setenv, and would be confusing because it's not just `SetEnv`, it also returns a cleanup function. `Override` doesn't really have any specific meaning in this context. 

Patch here, in the context of testing, refers to [Monkey Patching](https://en.wikipedia.org/wiki/Monkey_patch), not the more general [patch](https://en.wikipedia.org/wiki/Patch_(computing)) as in "partial change".

Reply:
Perhaps that's where my confusion was; anyway, it's not really related to this PR 😅